website/projects/aws/awssync_refactored.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+import logging
+
+from botocore.exceptions import ClientError
+
+from courses.models import Semester
+
+from projects.aws.awsapitalker import AWSAPITalker
+from projects.aws.awssync_structs import AWSTree
+
+
+class AWSSyncRefactored:
+    """Synchronise with Amazon Web Services."""
+
+    def __init__(self):
+        """Create an AWSSync instance."""
+        self.api_talker = AWSAPITalker()
+
+        self.logger = logging.getLogger("django.aws")
+        self.logger.setLevel(logging.DEBUG)
+
+    def create_course_ou(self, tree: AWSTree) -> str:
+        """Create organizational unit under root with name of current semester."""
+        root_id = tree.ou_id
+        course_ou_name = str(Semester.objects.get_or_create_current_semester())
+        course_ou_id = next((ou.ou_id for ou in tree.iterations if ou.name == course_ou_name), None)
```

Comment:
Name `matching_ous` suggests it can hold more than one element. We know from `Checks.check_double_iteration_names` that the resulting list will be either empty or have one element. The `[0]` access is also less elegant. 

It really comes down to the same thing at the end of the day and I do not think either solution has a significant difference in code readability. I suggest we keep it as is and focus on more important stuff. If you insist, we can do quick second opinion poll next meeting with others to reach consensus.

Reply:
It is indeed less pretty, but in my opinion more explicit for people that don't know the `next` function. But I'm fine with keeping your solution, just wanted to mention it.